### ISSUE TITLE
Negative trait impl gives wrong error message

### ISSUE THREAD

**Issue by DutchGhost** [rust-lang/rust#70849]
The following code:
```Rust
#![feature(negative_impls)]

struct MyType;

impl !Clone for &mut MyType {}

fn main() {}
```
produces:
```
  Compiling playground v0.0.1 (/playground)
error[E0119]: conflicting implementations of trait `std::clone::Clone` for type `&mut MyType`:
 --> src/main.rs:5:1
  |
5 | impl !Clone for &mut MyType {}
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: conflicting implementation in crate `core`:
          - impl<T> std::clone::Clone for &mut T
            where T: ?Sized;

error: aborting due to previous error

For more information about this error, try `rustc --explain E0119`.
error: could not compile `playground`.

To learn more, run the command again with --verbose.
```
which reads as: any `&mut T where T: ?Sized implements Clone`, which isn't realy true. It realy implements `!Clone`.

Perhaps the error message should add a `!` in front of the impl to make it more obvious, like this:
```
= note: conflicting implementation in crate `core`:
          - impl<T> !std::clone::Clone for &mut T
            where T: ?Sized;
```

**Comment by heiseish** [comment]
Hi @fmease , I'd like to take up this issue if it has not been assigned yet. However this would be my first contribution so I just want to confirm it's a good first task. Thanks!

**Comment by fmease** [comment]
Yeah, for sure! It should be as simple as updating `to_pretty_impl_header` ([see](https://github.com/rust-lang/rust//blob/842d6fc32e3d0d26bb11fbe6a2f6ae2afccc06cb/compiler/rustc_trait_selection/src/error_reporting/traits/mod.rs#L324)) to account for the polarity. You can query/obtain the impl polarity with `tcx.impl_polarity(impl_def_id)`. Feel free to reach out here or on Zulip if you have any questions.
